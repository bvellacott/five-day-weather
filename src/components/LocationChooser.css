.LocationChooser {
	background-color: #004;
}

.LocationChooser h1 {
	width: 100%;
	height: 100%;
	font-size: 4rem;
	color: #f9f3e0;
}

.LocationChooser h1 {
	background-color: #016;
}

.LocationChooser ul {
	position: relative;
	width: 100%;
	border-radius : 0rem 0rem .5rem .5rem;
	box-shadow: -0.075rem 0.075rem 0.05rem #444444;;
}

.LocationChooser li {
	font-size: 4rem;
	background-color: #004;
}

.LocationChooser li:nth-child(even) {
	background-color: #005;
}

.LocationChooser li:last-child {
	border-radius : 0rem 0rem .5rem .5rem;
}

.LocationChooser li {
	background-color: #016;
}

