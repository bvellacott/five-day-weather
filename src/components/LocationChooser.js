import React, { Component } from 'react';
import './LocationChooser.css';

function Title(props) {
  return <h1 onClick={props.setSearching} >{props.location}</h1>;
}

class Result extends Component {
	constructor(props) {
		super(props);
    this.setLocation = this.setLocation.bind(this);
	}

	setLocation() {
		this.props.setLocation(this.props.value);
	}

	render() {
		return (
			<span onClick={this.setLocation}>{this.props.value}</span>
		);
	}
}

class LocationChooser extends Component {
  constructor(props) {
    super(props);
    this.state = {
   		searching: false
    };

    this.setLocation = this.setLocation.bind(this);
    this.setSearching = this.setSearching.bind(this);
  }

  setLocation(newLoc) {
  	this.props.setLocation(newLoc);
  	this.setState({ searching: false });
  }

  setSearching() {
  	this.setState({ searching: true });
  }

  render() {
    return (
    	<div className="LocationChooser">
    		<Title location={this.props.location} searching={this.state.searching} setSearching={this.setSearching}/>
				{
					this.state.searching ? 
					<ul>
						{this.props.options.map( r => <li key={r}><Result value={r} setLocation={this.setLocation} /></li> )}
					</ul>
					: null
				}
      </div>
    );
  }
}

export default LocationChooser;
