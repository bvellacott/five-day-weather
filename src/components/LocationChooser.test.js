import React from 'react';
import LocationChooser from './LocationChooser';
import renderer from 'react-test-renderer';

test('Location set', () => {
  const component = renderer.create(
    <LocationChooser options={['a', 'b', 'c']} location='a' setLocation={function(newLoc) { expect(newLoc).toEqual('a'); }}/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[0].props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[1].children[0].children[0].props.onClick({ target: {} });
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
