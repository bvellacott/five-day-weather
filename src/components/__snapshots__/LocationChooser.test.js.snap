exports[`test Location set 1`] = `
<div
  className="LocationChooser">
  <h1
    onClick={[Function]}>
    a
  </h1>
</div>
`;

exports[`test Location set 2`] = `
<div
  className="LocationChooser">
  <h1
    onClick={[Function]}>
    a
  </h1>
  <ul>
    <li>
      <span
        onClick={[Function]}>
        a
      </span>
    </li>
    <li>
      <span
        onClick={[Function]}>
        b
      </span>
    </li>
    <li>
      <span
        onClick={[Function]}>
        c
      </span>
    </li>
  </ul>
</div>
`;

exports[`test Location set 3`] = `
<div
  className="LocationChooser">
  <h1
    onClick={[Function]}>
    a
  </h1>
</div>
`;
